snuba/util.py

```diff
@@ -69,6 +70,70 @@ def time_expr(alias, granularity):
 
     return template.format(column=column, granularity=granularity)
 
+def function_expr(fn, args_expr=''):
+    """
+    Generate an expression for a given function name and an already-evaluated
+    args expression. This is a place to define convenience functions that evaluate
+    to more complex expressions.
+
+    """
+    # For functions with no args, (or static args) we allow them to already
+    # include them as part of the function name, eg, "count()" or "sleep(1)"
+    if not args_expr and fn.endswith(')'):
+        return fn
+
+    # Convenience topK function eg "top10", "top3" etc.
+    topk = TOPK_FUNCTION_RE.match(fn)
+    if topk:
+        return 'topK({})({})'.format(topk.group(1), args_expr)
+
+    # turn uniq() into ifNull(uniq(), 0) so it doesn't return null where
+    # a number was expected.
+    if fn == 'uniq':
+        return 'ifNull({}({}), 0)'.format(fn, args_expr)
+
+    # emptyIfNull(col) is a simple pseudo function supported by Snuba that expands
+    # to the actual clickhouse function ifNull(col, '') Until we figure out the best
+    # way to disambiguate column names from string literals in complex functions.
+    if fn == 'emptyIfNull' and args_expr:
+        return 'ifNull({}, \'\')'.format(args_expr)
+
+    # default: just return fn(args_expr)
+    return  u'{}({})'.format(fn, args_expr)
+
+def is_function(column_expr, depth=0):
+    """
+    Returns a 3-tuple of (name, args, alias) if column_expr is a function,
+    otherwise None.
+
+    A function expression is of the form:
+
+        [func, [arg1, arg2]]  => func(arg1, arg2)
```

Comment:
@fpacifici is this clear? I feel like this would cover your question https://github.com/getsentry/snuba/pull/315/files/fe04817331d03839f1401e5154244a3c6e31235e#diff-f8bdb7ed98d8383aaabc8a4e54e49d0aR116

Reply:
Sorry I did not notice that one. Yes sure, that works